### ISSUE TITLE
Wondering if there's GPU version of pyroomacoustics or any method to accelerate

### ISSUE THREAD

**Issue by AlanLiudx** [LCAV/pyroomacoustics#267]
Hi @fakufaku. I'm getting more familiar with pra now and using it to generate a set of multi-channel noisy waveform with a number of 40000. Sadly, it might cost about 24 hours or so to generate the data. I wonder if there's any option for simulation acceleration or GPU version available?

**Comment by fakufaku** [comment]
Hi @AlanLiudx , there is no GPU support at the moment in pyroomacoustics, but you have a couple of options.
1) Parallelize: Use the python multi-processing module to generate RIR in parallel.
2) There is an alternative library [gpuRIR](https://github.com/DavidDiazGuerra/gpuRIR) that lets you use GPU

**Comment by AlanLiudx** [comment]
Thanks. I have already known gpuRIR but sadly I cannot manage to handle the environment problem to install gpuRIR on my server. I'll try multiprocessing instead.

**Comment by fakufaku** [comment]
I see! To get the best performance with multiprocessing it is important to reduce the number of threads used by numpy. On linux you can do it by setting the environment variable `OMP_NUM_THREADS=1`. If you use conda with MKL, then `MKL_NUM_THREADS=1`. This avoids each process over-subscribing threads.
In my experience, generating 40000 RIR should take a few hours (although it depends on the number of cores available in your machine).

**Comment by AlanLiudx** [comment]
Yes. Actually I know little about multi-process and thread.(sigh) There will be many problems to face with.

**Comment by fakufaku** [comment]
This is a pretty involved example, but I have used pyroomacoustics and multiprocessing to create a dataset in this repo: https://github.com/fakufaku/create_wsj1_2345_db

I will close the issue for now, but feel free to open another issue if you have problemes related to pyroomacoustics/multiprocessing.
